### PR TITLE
fix: extract and refactor yarn command (IN-304)

### PR DESCRIPTION
**Pull request by Xavientois**
**Fixes or implements IN-304**

### Brief description. What is this change?

Extract and refactor `yarn_command`

### Tests
Run outside of container: https://app.circleci.com/pipelines/github/voiceflow/platform/9205/workflows/48a14c47-9d37-4716-9f86-ee1be176e8ab/jobs/35772?invite=true#step-111-8
Run in container: https://app.circleci.com/pipelines/github/voiceflow/general-runtime/7060/workflows/05498e63-89f6-4ec0-ab01-52c51cae9d52/jobs/16223?invite=true#step-110-29
Parallelism with lock file: https://app.circleci.com/pipelines/github/voiceflow/general-runtime/7060/workflows/05498e63-89f6-4ec0-ab01-52c51cae9d52/jobs/16224

### CODE REVIEW

File: src/commands/yarn/yarn_command.yml [hunk 1]
```diff
@@ -69,49 +69,29 @@ steps:
...
+      condition: << parameters.run_in_container >>
+      steps:
+        - run:
+            name: Run << parameters.step_name >>
+            working_directory: << parameters.working_directory >>
+            background: << parameters.run_in_background >>
+            command: |
+              trap 'echo "fail detected"; touch /tmp/failure' ERR
+              << parameters.yarn_command >>
+              <<#parameters.run_in_background>>rm -rf ${LOCK_FILE?}<</parameters.run_in_background>>
```
Comment (xavidop) [inline src/commands/yarn/yarn_command.yml:82]: add an if-else statement here

**Review by xavidop — COMMENTED** [review]
can you test this in a monorepo with and without a container?

**Comment by Xavientois** [comment]
Monorepo outside of container: https://app.circleci.com/pipelines/github/voiceflow/platform/9225/workflows/a09d7168-0c7a-4827-a093-4a70152113dc/jobs/35907?invite=true#step-117-14

Monorepo inside container: https://app.circleci.com/pipelines/github/voiceflow/creator-app/154419/workflows/3ab54612-6c02-4ec4-90a3-d01c57615bb9/jobs/630406?invite=true#step-110-29

File: src/scripts/yarn/run_command.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+# shellcheck disable=SC2086
+
+trap 'echo "fail detected"; touch /tmp/failure' ERR
+
+echo "Running command: ${COMMAND:?}"
+bash -c "${COMMAND:?}"
```
Comment (xavidop) [inline src/scripts/yarn/run_command.sh:7]: if the command fails, is this catch by the trap? Im wondering it because it is running in a sub-shell

**Review by xavidop — CHANGES_REQUESTED** [review]

File: src/scripts/yarn/run_command.sh (hunk 2 above)
Comment (Xavientois) [reply src/scripts/yarn/run_command.sh:7]: Yup!
```
bash-3.2$ trap 'echo "fail detected"; touch /tmp/failure' ERR
bash-3.2$ bash -c "exit 1"
fail detected
bash-3.2$
```

Comment (xavidop) [reply src/scripts/yarn/run_command.sh:7]: NOICE

**Review by xavidop — APPROVED** [review]
love the change!

**Pull request merged**